staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+// ContainerType signifies container type
+type ContainerType int
+
+const (
+	// Containers is for normal containers
+	Containers ContainerType = 1 << iota
+	// InitContainers is for init containers
+	InitContainers
```

Comment:
Should we mention that the current init containers includes sidecar containers in the comment? Not have to. The pod limit\request calculation session has already include the KEP link below. https://github.com/kubernetes/kubernetes/pull/124609/files#diff-22767ad86bae6a15ac904d98655b9f1eff79bf51470b419c85c1d09d41cdb011R108

Reply:
This relates to the question I left on the other PR https://github.com/kubernetes/kubernetes/pull/127878#issuecomment-2405521951

Should we be worried about other users of ContainerFn. Maybe it's worth adding SidecarContainers before putting this method in the helpers package?

Reply:
I think it's better add `SidecarContainers` when some developer actually needs to use it?

Reply:
https://github.com/kubernetes/kubernetes/pull/116429#discussion_r1235351189

@smarterclayton and I had a similar discussion.

I am worried about explicitly using SidecarContainer naming (other than the feature name), as the "sidecar" is quite a general term.

I think the point of introducing the "Always" containerRestartPolicy was not to create one concrete "SidecarContainer" type. It just updated the restart behavior of the existing init containers so that kubernetes can have native way to support sidecar **pattern**.

Reply:
I wouldn't change this here / now in this PR

sidecar containers *are* init containers, just of a special type... if we want to clarify that later in a comment, that could be fine

Reply:
> Should we be worried about other users of ContainerFn? Maybe it's worth adding SidecarContainers before putting this method in the helpers package?

I think this PR should be a *strictly* mechanical move, not change semantics